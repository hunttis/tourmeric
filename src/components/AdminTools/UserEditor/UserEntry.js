import React from 'react';
import PropTypes from 'prop-types';
import { Translate } from 'react-localize-redux';

export const UserEntry = ({ openEditModal, openCreditModal, openDeleteModal, userData, creditAmount }) => (
  <div className="userentry column is-half is-one-third-desktop is-one-third-widescreen is-one-quarter-fullhd">
    <div className="card">

      <div className="card-header">
        <div className="card-header-title">
          {userData.displayName} - {userData.email}
        </div>
        {userData.avatarUrl &&
        <div className="card-header-icon">
          <figure className="image is-48x48">
            <img src={userData.avatarUrl} alt="User avatar" />
          </figure>
        </div>
      }
      </div>

      <div className="card-content">
        <ul>
          {userData.displayName &&
            <li><Translate id="displayname" />: {userData.displayName}</li>
          }
          {userData.username &&
            <li><Translate id="username" />: {userData.username}</li>
          }
          <li><Translate id="email" />: {userData.email}</li>
          {userData.avatarUrl &&
            <li><Translate id="avatarurl" />:&nbsp;
              <a target="_blank" rel="noopener noreferrer" href={userData.avatarUrl}>{userData.avatarUrl.substr(0, 20)}...</a>
            </li>
          }
          {userData.dciNumber &&
            <li><Translate id="dcinumber" />: {userData.dciNumber}</li>
          }
          <li><Translate id="storecredit" />: <span className="has-text-success">{creditAmount} â‚¬</span></li>
        </ul>
      </div>
      <div className="card-footer">
        <div className="card-footer-item button is-primary" onClick={openEditModal}>
          <Translate id="edit" />
        </div>
        <div className="card-footer-item button is-info" onClick={openCreditModal}>
          <Translate id="storecredit" />
        </div>
        <div className="card-footer-item button is-danger" onClick={openDeleteModal}>
          <Translate id="delete" />
        </div>
      </div>
    </div>
  </div>
);

UserEntry.propTypes = {
  openEditModal: PropTypes.func,
  openDeleteModal: PropTypes.func,
  openCreditModal: PropTypes.func,
  userData: PropTypes.object,
  creditAmount: PropTypes.number,
};
