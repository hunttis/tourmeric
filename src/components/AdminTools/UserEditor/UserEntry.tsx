import React from 'react';
import { Translate } from 'react-localize-redux';

import CreditAmounts from './CreditAmounts-container';
import { User } from '~/models/ReduxState';

interface Props {
  openEditModal: () => void;
  openDisableModal: () => void;
  openCreditModal: () => void;
  userId: string;
  userData: User;
}

export const UserEntry = ({ openEditModal, openCreditModal, openDisableModal, userData, userId }: Props) => (
  <div className="userentry column is-half is-one-third-desktop is-one-third-widescreen is-one-quarter-fullhd">
    <div className="card">

      <div className="card-header">
        <div className="card-header-title">
          {userData.firstName} {userData.lastName} - {userData.email}
        </div>
        {userData.avatarUrl &&
          <div className="card-header-icon">
            <figure className="image is-48x48">
              <img src={userData.avatarUrl} alt="User avatar" />
            </figure>
          </div>
        }
      </div>

      <div className="card-content">
        <ul>
          {userData.firstName &&
            <li><Translate id="firstname" />: {userData.firstName}</li>
          }
          {userData.lastName &&
            <li><Translate id="lastname" />: {userData.lastName}</li>
          }
          {userData.username &&
            <li><Translate id="username" />: {userData.username}</li>
          }
          <li><Translate id="email" />: {userData.email}</li>
          {userData.avatarUrl &&
            <li><Translate id="avatarurl" />:&nbsp;
              <a target="_blank" rel="noopener noreferrer" href={userData.avatarUrl}>{userData.avatarUrl.substr(0, 20)}...</a>
            </li>
          }
          {userData.dciNumber &&
            <li><Translate id="dcinumber" />: {userData.dciNumber}</li>
          }
          <li>&nbsp;</li>
          <li><strong><Translate id="storecredit" /></strong></li>
          <CreditAmounts userId={userId} />
        </ul>
      </div>
      <div className="card-footer">
        <div className="card-footer-item button is-primary" onClick={openEditModal}>
          <Translate id="edit" />
        </div>
        <div className="card-footer-item button is-info" onClick={openCreditModal}>
          <Translate id="storecredit" />
        </div>
        <div className="card-footer-item button is-danger" onClick={openDisableModal}>
          <Translate id="delete" />
        </div>
      </div>
    </div>
  </div>
);
