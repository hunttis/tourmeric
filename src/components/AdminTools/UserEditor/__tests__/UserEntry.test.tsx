import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { UserEntry } from '../UserEntry';

describe('UserEntry Tests', () => {

  it('Compares UserEntry to snapshot', () => {

    const mockUserData = {}; // Todo mock user data
    const mockOpenModal = () => {};

    const userEntry = shallow(<UserEntry
      userData={mockUserData}
      openEditModal={mockOpenModal}
      openDisableModal={() => {}}
      openCreditModal={() => {}}
      userId="testid-123"
    />);

    expect(userEntry.exists()).toBe(true);
    expect(toJson(userEntry)).toMatchSnapshot();

  });

  // TODO: Write tests for clicking the modals

});
