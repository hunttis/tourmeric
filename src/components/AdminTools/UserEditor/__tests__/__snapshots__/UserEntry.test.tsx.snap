// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserEntry Tests Compares UserEntry to snapshot 1`] = `
<div
  className="userentry column is-half is-one-third-desktop is-one-third-widescreen is-one-quarter-fullhd"
>
  <div
    className="card"
  >
    <div
      className="card-header"
    >
      <div
        className="card-header-title"
      >
         
         - 
      </div>
    </div>
    <div
      className="card-content"
    >
      <ul>
        <li>
          <Translate
            id="email"
          />
          : 
        </li>
        <li>
          Â 
        </li>
        <li>
          <strong>
            <Translate
              id="storecredit"
            />
          </strong>
        </li>
        <Connect(Connect(Component))
          userId="testid-123"
        />
      </ul>
    </div>
    <div
      className="card-footer"
    >
      <div
        className="card-footer-item button is-primary"
        onClick={[Function]}
      >
        <Translate
          id="edit"
        />
      </div>
      <div
        className="card-footer-item button is-info"
        onClick={[Function]}
      >
        <Translate
          id="storecredit"
        />
      </div>
      <div
        className="card-footer-item button is-danger"
        onClick={[Function]}
      >
        <Translate
          id="delete"
        />
      </div>
    </div>
  </div>
</div>
`;
